frontend/src/modules/_shared/components/ReadoutBox.tsx

```diff
@@ -0,0 +1,140 @@
+    const readoutItems = props.readoutItems;
+    const maxNumItems = props.maxNumItems ?? 3;
+
+    const [flipped, setFlipped] = useState<boolean>(false);
+    const readoutRoot = useRef<HTMLDivElement>(null);
+
+    // How far away from the lower right corner the box is placed
+    // TODO: Expose as prop?
+    // ! "3" is based on the left-12/right-12 layout values.
+    const cornerDistance = parseInt(getComputedStyle(document.documentElement).fontSize) * 3;
+
+    useEffect(() => {
+        if (props.flipDisabled) return;
+
+        function maybeFlipBox(evt: MouseEvent) {
+            if (!readoutRoot.current) return;
+
+            const offsetParent = readoutRoot.current.offsetParent;
+            if (!offsetParent) return; // Not floating, I believe
+
+            const parentRect = offsetParent.getBoundingClientRect();
+            const { top, bottom, width } = readoutRoot.current.getBoundingClientRect();
+
+            // If above, or below, it's guaranteed to fit
+            if (evt.clientY < top || evt.clientY > bottom) {
+                if (flipped) setFlipped(false);
+                return;
+            }
+
+            const prefferredRight = parentRect.right - cornerDistance;
+            const prefferredLeft = parentRect.right - width - cornerDistance;
+
+            if (evt.clientX < prefferredLeft || evt.clientX > prefferredRight) {
+                if (flipped) setFlipped(false);
+            } else {
+                if (!flipped) setFlipped(true);
+            }
+        }
+
+        document.addEventListener("mousemove", maybeFlipBox);
+
+        return () => document.removeEventListener("mousemove", maybeFlipBox);
+    }, [cornerDistance, flipped, props.flipDisabled]);
+
+    if (readoutItems.length === 0) return null;
+
+    return (
+        <div
+            ref={readoutRoot}
+            className={`absolute bottom-10 right-12 z-50 w-60  flex flex-col gap-2 p-2 text-sm rounded border border-neutral-300 bg-white bg-opacity-75 backdrop-blur-sm pointer-events-none ${
+                flipped ? "left-12" : "right-12"
+            }`}
+        >
+            {readoutItems.map((item, idx) => {
+                if (idx < maxNumItems) {
+                    return (
+                        <Fragment key={idx}>
+                            <div className="flex gap-2 font-bold items-center">
+                                {!props.noLabelColor && (
+                                    <div
+                                        className="rounded-full w-3 h-3 border border-slate-500"
+                                        style={{ backgroundColor: item.color }}
+                                    />
+                                )}
+                                <span className="block">{item.label}</span>
+                            </div>
+
+                            {item.info && item.info.map((i: InfoItem, idx: number) => <InfoItem key={idx} {...i} />)}
```

Comment:
`InfoItem`s are rendered as `table-rows` but there is no parent that renders as `table`. This could already have been an issue in the old code, I did not check, but good to have a proper implementation here.

Reply:
That is something from the old code. Is there any specific reason we're using these table classes instead of proper table tags? Regardless, I think it be more correct to use flex/grid displays here instead anyways

Reply:
I went for a grid solution. See: 1ac894b2c0229f2738d14f5de3c58001e66b937c